sphinx/writers/latex.py

```diff
@@ -121,29 +140,28 @@
         'inputenc':     '\\usepackage[utf8]{inputenc}',
         'utf8extra':   ('\\ifdefined\\DeclareUnicodeCharacter\n'
                         '% support both utf8 and utf8x syntaxes\n'
-                        '\\edef\\sphinxdqmaybe{'
-                        '\\ifdefined\\DeclareUnicodeCharacterAsOptional'
-                        '\\string"\\fi}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe00A0}'
-                        '{\\nobreakspace}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe2500}'
-                        '{\\sphinxunichar{2500}}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe2502}'
-                        '{\\sphinxunichar{2502}}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe2514}'
-                        '{\\sphinxunichar{2514}}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe251C}'
-                        '{\\sphinxunichar{251C}}\n'
-                        '  \\DeclareUnicodeCharacter{\\sphinxdqmaybe2572}'
-                        '{\\textbackslash}\n'
+                        '  \\ifdefined\\DeclareUnicodeCharacterAsOptional\n'
+                        '    \\def\\sphinxDUC#1{\\DeclareUnicodeCharacter{"#1}}\n'
+                        '  \\else\n'
+                        '    \\let\\sphinxDUC\\DeclareUnicodeCharacter\n'
+                        '  \\fi\n'
+                        '  \\sphinxDUC{00A0}{\\nobreakspace}\n'
+                        '  \\sphinxDUC{2500}{\\sphinxunichar{2500}}\n'
+                        '  \\sphinxDUC{2502}{\\sphinxunichar{2502}}\n'
+                        '  \\sphinxDUC{2514}{\\sphinxunichar{2514}}\n'
+                        '  \\sphinxDUC{251C}{\\sphinxunichar{251C}}\n'
+                        '  \\sphinxDUC{2572}{\\textbackslash}\n'
                         '\\fi'),
     },
     'xelatex': {
         'latex_engine': 'xelatex',
         'polyglossia':  '\\usepackage{polyglossia}',
         'babel':        '',
         'fontenc':      '\\usepackage{fontspec}',
-        'fontpkg':      '',
+        'fontpkg':     ('\\setmainfont{CMU Serif}\n'
+                        '\\setsansfont{CMU Sans Serif}\n'
+                        '\\setmonofont{CMU Typewriter Text}'),
```

Comment:
These fonts are visually close to previous default with `xelatex`:  previous default was LaTeX choice i.e. Latin Modern OpenType, which is not far from original Knuth Computer Modern. The CMU Serif is the Unicode version of Computer Modern, it exists in serif, sans serif, monospace and supports both Greek and Cyrillic.

Alternative is Libertinus, which is more Times-like (derived from Linux Libertine) but this is more recent font, currently still under development recently.